assets/tutorials/multiple_feasible_solutions.jl

```diff
@@ -0,0 +1,240 @@
+# Copyright (c) 2021 James D Foster, and contributors                            #src
+#                                                                                #src
+# Permission is hereby granted, free of charge, to any person obtaining a copy   #src
+# of this software and associated documentation files (the "Software"), to deal  #src
+# in the Software without restriction, including without limitation the rights   #src
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell      #src
+# copies of the Software, and to permit persons to whom the Software is          #src
+# furnished to do so, subject to the following conditions:                       #src
+#                                                                                #src
+# The above copyright notice and this permission notice shall be included in all #src
+# copies or substantial portions of the Software.                                #src
+#                                                                                #src
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     #src
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       #src
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    #src
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         #src
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  #src
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  #src
+# SOFTWARE.                                                                      #src
+
+# # Finding multiple feasible solutions 
+
+# **Author**: James Foster (@jd-foster)
+
+# This brief tutorial demonstrates how to formulate and solve a combinatorial problem
+# with multiple solutions. In fact, we will see how to find _all_ feasible solutions to our problem.
+# We will also see how to enforce an "all-different" constraint on a set of integer variables.
+
+# We start with some objects from recreational mathematics that could be called **"symmetric square sums"**.
+# Here are a few examples (inspired by [Number Squares](https://www.futilitycloset.com/2012/12/05/number-squares/)):
+"""
+    1529      2318      5219
+    5837      3790      2384
+    2340      1956      1867
+  + ____    + ____    + ____
+    9706      8064      9470
+"""
+
+# Notice how all the digits 0 to 9 are used at least once,
+# the first three rows sum to the last row,
+# and the columns in each are the same as the corresponding rows (forming a symmetric matrix).
```

Comment:
I wonder if we can format these differently, since it took me a while to figure out what was going on. I think it's just a matter of spacing the columns further apart, and getting rid of the `---` row?
```
   1 5 2 9
   5 8 3 7
 + 2 3 4 0
 = 9 7 0 6
```

Reply:
Good idea. Changed.